python/ray/data/datasource/file_based_datasource.py

```diff
@@ -511,10 +511,9 @@ def __init__(
                     "No input files found to read. Please double check that "
                     "'partition_filter' field is set properly."
                 )
-
-        ctx = DataContext.get_current()
-        shuffler_class = get_attribute_from_class_name(ctx.file_metadata_shuffler)
-        self._file_metadata_shuffler = shuffler_class(self._reader_args)
+        self._file_metadata_shuffler = FileMetadataShuffler(
+            reader_args.get("shuffle", None)
+        )
```

Comment:
I feel the implicit `reader_args` really hurts readability. it's hard to find the `shuffle` argument is eventually used here. I'd suggest using an explicit argument if possible. 
Also, for the api, it's okay to support literal strings `shuffle="files"`, but it'd improve readability and maintainability to define a class `ShuffleOptions` internally.

Reply:
+1 `reader_args` is confusing. Should be addressed with the `Datasource` refactors